nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/RotationalDynamics.java

```diff
@@ -200,12 +200,12 @@ public void advanceDihedrals2(double timestep) {
                 max = absDelta;
             }
             sumSq += delAngle * delAngle;
-            //System.out.println(deltaSum);
             if (daughter == null) {
-                System.out.println("daughter atom is null " + diAtom.getShortName());
+                throw new NullPointerException("daughter atom is null " + diAtom.getShortName());
```

Comment:
This print statement was added specifically for debugging but nothing done about the NPE so assumed it is meant to be thrown here and moved the print statement to the exception message.

Reply:
Used Objects.requireNonNull